src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/WeakEventTable.cs

```diff
@@ -108,7 +108,11 @@ internal IDisposable WriteLock
         /// </summary>
         internal WeakEventManager this[Type managerType]
         {
-            get { return (WeakEventManager)_managerTable[managerType]; }
+            get
+            {
+                _managerTable.TryGetValue(managerType, out WeakEventManager result);
+                return result;
+            }
             set { _managerTable[managerType] = value; }
```

Comment:
I think that a remove should be done if value is null.

Reply:
you mean the result variable?
I think it is never null, when key exitst in the Hashset.
TryGetValue is needed, because the key maybe does not exists in the Dictionary. (Hashset returned null in this case, Dictionary throws exception when I use _managerTable[managerType])

Reply:
I am talking about the set. What if the user passes "null" as the value?
We shouldn't keep a key/value pair in the dictionary for a null value.

Reply:
Ok, so in the setter... I did not want to make any functional change, but for me that is OK :)
This setter is called from `WeakEventManager.SetCurrentManager`, which is protected, so accessible for users (not internal). I think that method should check the argument.

Reply:
OK... sounds good to me.